module3/lessons/Week2/IntroToMVC.md

```diff
@@ -0,0 +1,307 @@
+
+<aside class="instructor-notes">
+    <p><strong>Instructor Note</strong><br>
+    Use this powerpoint for the start and end of the lesson. https://docs.google.com/presentation/d/1YKc7fF2SyeWJUmmw2Elw4HujCgkRUVbwuE9SXzimqdo/edit?usp=sharing
+    <br>It's set up to use Pear Deck, because think that fits very well here, but this can totally be done without pear deck to avoid another thing to keep track of with this lesson!<br> There is a branch in the starter code repo called finishedMVC that has the finished version of today's lesson. 
+    </p>
+</aside>
+
+
+Today is a big day in your developer journey! We're going to put all of the pieces together and build a full application using the MVC pattern.
+
+Here is our plan:
+
+1. Get Starter Project Running
+1. Add a Controller
+1. Add a View
+1. Controller and View Practice
+1. Add a Model
+1. Connect the M, V, and C
+1. Zoom Back Out and Draw a Diagram
+
+## 1. Get Starter Project Running
+
+For today's lesson, we are going to build off of some starter code. MVC is the focus of today, not the starter code, so don't worry about what's already there.
+
+Let's all fork and clone this repo: https://github.com/turingschool-examples/MvcMovieStarter
+
+If you're curious, this is what has been done so far:
+
+1. Use the ASP.NET Core Web App (Model-View-Controller) template to create a new project
+1. Install the following packages
+    * Npgsql.EntityFrameworkCore.PostgreSQL
+    * Microsoft.EntityFrameworkCore.Tools
+    * EFCore.NamingConventions
+1. Configure database connection string in appsettings.json
+1. Add dbContext in DataAccess/MVCMoviesContext.cs
+1. Register the context in Program.cs using the database connection string and snake_case naming convention
+
+To run this application, click the outlined "play" button to "Start Without Debugging". You will probably see the following confirmations
+
+![Creating a Web App Confirmation Messages](/assets/images/module3/Week2/MVC_2_Create_Web_App_Confirmations.png)
+
+When you click start, Visual Studio runs the app and opens it in the default browser. You should see a page that says "Welcome Learn about building Web apps with ASP.NET Core."
+
+The address bar shows `localhost:<port#>`. Localhost means the application is running on your local computer. When Visual Studio creates a web project, a random port is used for the web server.
+
+## 2. Add a Controller
+
+<aside class="instructor-notes">
+    <p><strong>Instructor Note</strong><br>I encourage referencing back to the sides from the start of class to refresh what a controller is before diving into this section. Same thing for the View and Model.
+    </p>
+</aside>
+
+We're going to use the scaffolding tool in Visual Studio to create our controller. Follow the instructions in the following image, except call your controller `MoviesController.cs` instead of `HelloWorldController.cs`
+
+![Creating a Controller](/assets/images/module3/Week2/MVC_3_Create_Controller.png)

```

Comment:
Is it possible to create new screenshots for creating Controller instead of a note that says `"...except call your controller MoviesController.cs instead of HelloWorldController.cs"`?

Another option would be to cut off the "HelloWorldController.cs" part of the existing image and add the appropriate text instead.

Reply:
The screenshots are from the microsoft tutorials, and they have the nice steps written out which is why I was hesitant to make my own. I can see how this would be confusing and with so much new stuff want to remove as much confusing as I can, thanks for the note. Maybe for now I'll add a text box over where it says HelloWorldController.